File: 1512. Number of Good Pairs.cpp
class Solution
{
public:
    int numIdenticalPairs(vector<int> &nums)
    {
        int h[101] = {
            0,
        };

        int h_idx = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            h_idx = nums[i];
            h[h_idx]++;
        }

        int res = 0;
        for (int i = 0; i < 101; i++)
        {
            if (h[i] > 0)
            {
                res = res + (h[i] * (h[i] - 1) / 2);
            }
        }
        return res;
    }
};

File: 1512. Number of Good Pairs.py
class Solution(object):
    def numIdenticalPairs(self, nums):
        h = [0] * 101
        l_num = set()
        for i in range(0, len(nums)):
            h_idx = nums[i]
            h[h_idx] = h[h_idx] + 1
            l_num.add(h_idx)

        res = 0
        for idx in l_num:
            num = h[idx]
            if num > 1:
                res = res + int(num * (num - 1) / 2)

        return res